lib/charms/layer/kubernetes_common.py

```diff
@@ -933,3 +934,199 @@ def get_secret_password(username):
+        feature_gates["RotateKubeletServerCertificate"] = True
+    if is_state("kubernetes-worker.gpu.enabled"):
+        feature_gates["DevicePlugins"] = True
+    if feature_gates:
+        kubelet_config["featureGates"] = feature_gates
+    if is_dual_stack(cluster_cidr()):
+        feature_gates = kubelet_config.setdefault("featureGates", {})
+        feature_gates["IPv6DualStack"] = True
+
+    # Workaround for DNS on bionic
+    # https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/655
+    resolv_path = os.path.realpath("/etc/resolv.conf")
+    if resolv_path == "/run/systemd/resolve/stub-resolv.conf":
+        kubelet_config["resolvConf"] = "/run/systemd/resolve/resolv.conf"
+
+    # Add kubelet-extra-config. This needs to happen last so that it
+    # overrides any config provided by the charm.
+    kubelet_extra_config = hookenv.config("kubelet-extra-config")
+    kubelet_extra_config = yaml.safe_load(kubelet_extra_config)
+    merge_kubelet_extra_config(kubelet_config, kubelet_extra_config)
+
+    # Render the file and configure Kubelet to use it
+    os.makedirs("/root/cdk/kubelet", exist_ok=True)
+    with open("/root/cdk/kubelet/config.yaml", "w") as f:
+        f.write("# Generated by kubernetes-worker charm, do not edit\n")
+        yaml.dump(kubelet_config, f)
+    kubelet_opts["config"] = "/root/cdk/kubelet/config.yaml"
+
+    # If present, ensure kubelet gets the pause container from the configured
+    # registry. When not present, kubelet uses a default image location
+    # (currently k8s.gcr.io/pause:3.4.1).
+    if registry:
+        kubelet_opts["pod-infra-container-image"] = get_sandbox_image_uri(registry)
+
+    if taints:
+        kubelet_opts["register-with-taints"] = ",".join(taints)
+
+    workaround_lxd_kernel_params()
+
+    configure_kubernetes_service(
+        "kubernetes-common.prev-args.", "kubelet", kubelet_opts, "kubelet-extra-args"
+    )
+
+
+def configure_default_cni(default_cni):
+    """Set the default CNI configuration to be used by CNI clients
+    (kubelet, containerd).
+
+    CNI clients choose whichever CNI config in /etc/cni/net.d/ is
+    alphabetically first, so we accomplish this by creating a file named
+    /etc/cni/net.d/05-default.conflist, which is alphabetically earlier than
+    typical CNI config names, e.g. 10-flannel.conflist and 10-calico.conflist
+
+    The created 05-default.conflist file is a symlink to whichever CNI config
+    is actually going to be used.
+    """
+    # Clean up current default
+    cni_conf_dir = "/etc/cni/net.d"
+    for filename in os.listdir(cni_conf_dir):
+        if filename.startswith("05-default."):
+            os.remove(cni_conf_dir + "/" + filename)
+
+    # Set new default
+    cni = endpoint_from_flag("cni.available")
+    cni_conf = cni.get_config(default=default_cni)
+    source = cni_conf["cni-conf-file"]
+    dest = cni_conf_dir + "/" + "05-default." + source.split(".")[-1]
+    os.symlink(source, dest)
```

Comment:
Removed from [here](https://github.com/charmed-kubernetes/charm-kubernetes-worker/pull/103/files#diff-1837fd4ad476d5c87c628d68b3c461fc5c03d6aeafc928e910c69aa6ae267f85L760-L872) with a change of signature from

```python
def configure_default_cni():
   ...
```

to

```python
def configure_default_cni(default_cni):
   ...
```

Reasons:
* `default_cni` - added because of a loss of scope, provided now by worker or master